### PR TITLE
[Five-330] (Frontend) Añadir edición de artefactos custom

### PR DESCRIPTION
**Pull request by gperelstein**
Este PR corresponde a la parte del frontend de la modificación de artefactos custom. Se hizo un formulario parecido al de creación donde se puede modificar el nombre del artefacto y las settings (ya sea nombre o valor). Al modificarse el nombre de un setting se borran todas las relaciones asociadas a esa setting

https://user-images.githubusercontent.com/71278846/109206441-9db2dd00-7786-11eb-8a41-5121584fa339.mp4

### CODE REVIEW

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            } else {
+                props.setSnackbarSettings({ message: "Hubo un error al modificar el artefacto", severity: "error" });
+                props.setOpenSnackbar(true);
+                return;
+            }
+        }
+        catch (error) {
+            props.setSnackbarSettings({ message: "Hubo un error al crear el artefacto", severity: "error" });
+            props.setOpenSnackbar(true);
+        }
```
Comment (gperelstein) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx:74]: I was thinking that the delete of relations could be done in the backend

Comment (matiasbeckerle) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx:74]: If the operation should be considered as one, then that's probably a good idea. If they're independent then maybe not.

**Review by matiasbeckerle — APPROVED** [review]

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx (hunk 1 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx]: Yes, it should be considered as a single operation. I moved it to the artifact service #112

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifact.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+                clearErrors(`settings[${array[0]}].name`);
+            }
+        }
+    }
+
+    //Check if there are names repeated in settingsMap
+    const areNamesRepeated = (index: number) => {
+        let key = searchIndexInObject(settingsMap, index);
+        if (key != null && (settingsMap[key].length > 1 || key === 'price')) {
+            console.log("Hola");
```
[the remark this answers is not in the document]
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifact.tsx]: I'm very sorry about this :disappointed:. I have to get used to checking eslint errors before creating the PRs. I will work on this. I already deleted the console.log()

[the remark this answers is not in the document]
Comment (matiasbeckerle) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifact.tsx]: Lol don't worry, only take care of it. You can consider performing a lint validation as a github action or include it as a step in the tooling you already have in place? That way you never forget again: computers will take care and let you know if you forgot.

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                                return (<ul>{name}</ul>);
+                            })}
+                        </li>
+                    </Typography> :
+                    null
+                }
+                <hr/>
+                <Typography gutterBottom>
+                    Revise las características con las que quedará su artefacto modificado
+                    y se está de acuerdo haga click en confirmar
```
Comment (martinmena) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx:95]: I think you meant "_y **si** está de acuerdo_"

**Review by martinmena — APPROVED** [review]

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx (hunk 3 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifactConfirmation.tsx]: It is fixed!

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifact.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/ArtifactsComponents/EditArtifact.tsx]: It's a great idea to add an automatic validation for this! I will look for alternatives

**Pull request merged**